### PR TITLE
misc dep bumps

### PR DESCRIPTION
**Pull request by turadg**
evergreen

## Description
Bump a bunch of deps and also dedupe.

- make multichain-testing use `dev` head of cosmic-proto (solving a warning)
- Deduping Endo in multichain-testing 
- bump typescript-eslint should solve the TS version warning

### Security Considerations
none

### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
CI

### Upgrade Considerations
none

### CODE REVIEW

File: multichain-testing/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   },
   "packageManager": "yarn@4.5.3",
   "devDependencies": {
-    "@agoric/cosmic-proto": "0.4.1-dev-08f8549.0",
+    "@agoric/cosmic-proto": "dev",
```
Comment (mhofman) [inline multichain-testing/package.json:20]: I suppose `latest` is not an option (aka the upgrade-18 packages are not up to date enough?)

File: packages/eslint-config/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   "peerDependencies": {
     "@endo/eslint-plugin": "^2.2.3",
     "@jessie.js/eslint-plugin": "^0.4.1",
-    "typescript-eslint": "^7.13.1",
+    "typescript-eslint": "^8.17.0",
```
Comment (mhofman) [inline packages/eslint-config/package.json:30]: We're getting a lot of duplication because of this major bump. Any way to get rid of the v7 in any way? Looks like the only dependency is `eslint-plugin-github` whose [latest version](https://github.com/github/eslint-plugin-github/blob/v5.1.3/package.json#L38) has bumped to 8 as well.

**Review by mhofman — COMMENTED** [review]
Please consider bumping `eslint-plugin-github` to remove some new duplication of `typescript-eslint` packages

File: multichain-testing/package.json (hunk 1 above)
Comment (turadg) [reply multichain-testing/package.json:20]: `0.5.0-u18` may work but I think it's better to integrate with head

File: packages/eslint-config/package.json (hunk 2 above)
Comment (turadg) [reply packages/eslint-config/package.json:30]: Good idea. Bumping that bumped acorn, so I had to forward the patch. Hopefully for the last time ,
- https://github.com/acornjs/acorn/issues/1333

**Review by mhofman — APPROVED** [review]

File: .eslintrc.cjs [hunk 3]
```diff
@@ -62,15 +62,12 @@ module.exports = {
...
-    project: [
-      './packages/*/tsconfig.json',
-      './packages/*/tsconfig.json',
-      './packages/wallet/*/tsconfig.json',
-      './tsconfig.json',
-    ],
+    projectService: {
+      allowDefaultProject: ['*.js'],
+      defaultProject: 'tsconfig.json',
+    },
```
Comment (mhofman) [inline .eslintrc.cjs:70]: Why is this necessary? From what I understand of the [eslint parser config](https://typescript-eslint.io/packages/parser/#allowdefaultproject), if the `tsconfig.json` already has `allowJs` / `checkJs` true, then it will include `.js` files. And since `defaultProject` already defaults to `tsconfig.json`, the defaults should work just as well?

```suggestion
    projectService: true,
```

**Review by mhofman — APPROVED** [review]
Small question on eslint parser config, but besides that LGTM

File: .eslintrc.cjs (hunk 3 above)
Comment (turadg) [reply .eslintrc.cjs:70]: I'm not certain that it's necessary now. Some version of it was necessary at some point in getting it all to work. I think it's fine to leave for the defaults visible.

Comment (mhofman) [reply .eslintrc.cjs:70]: I read about some negative performance impact of including too many files that are not part of the project, that's why I asked if it was necessary.

Comment (turadg) [reply .eslintrc.cjs:70]: Ah. There are very few that aren't part of a package project. And if they're not part of any project the linter fails. That's why I had to exclude exported.*, etc from types linting.

Fwiw, [lint-primary for this PR took 13m10s](https://github.com/Agoric/agoric-sdk/actions/runs/12201510884/job/34040236423) and [13m35s after merging to master](https://github.com/Agoric/agoric-sdk/actions/runs/12202220310/job/34042581059). [Master right before that was 13m34s](https://github.com/Agoric/agoric-sdk/actions/runs/12199535025/job/34033765021), so basically no change.
